lib/zendesk_api/pagination.rb

```diff
@@ -0,0 +1,107 @@
+module ZendeskAPI
+  # Contains all methods related to pagination in an attempt to slim down collection.rb
+  module Pagination
+    DEFAULT_PAGE_SIZE = 100
+    def more_results?(response)
+      Helpers.present?(response["meta"]) && response["meta"]["has_more"]
+    end
+    alias has_more_results? more_results? # For backward compatibility with 1.33.0 and 1.34.0
+
+    def each_page!(*args, &block)
+      warn "ZendeskAPI::Collection#each_page! is deprecated, please use ZendeskAPI::Collection#all!"
+      all!(*args, &block)
+    end
+
+    def each_page(*args, &block)
+      warn "ZendeskAPI::Collection#each_page is deprecated, please use ZendeskAPI::Collection#all"
+      all(*args, &block)
+    end
+
+    # Changes the per_page option. Returns self, so it can be chained. No execution.
+    # @return [Collection] self
+    def per_page(count)
+      clear_cache if count
+      @options["per_page"] = count
+      self
+    end
+
+    # Changes the page option. Returns self, so it can be chained. No execution.
+    # @return [Collection] self
+    def page(number)
+      clear_cache if number
+      @options["page"] = number
+      self
+    end
+
+    def first_page?
+      !@prev_page
+    end
+
+    def last_page?
+      !@next_page || @next_page == @query
+    end
+
```

Comment:
Moving public methods make is a bit harder to follow in case they're called from outside of Collection, but in this case I'm fine with it. 👍